File: Skynet-P3-Frontend-main/src/models/Post.ts
import Comment from './Comment';

export default class Post {
  postId: number;
  text: string;
  imageUrl: string;
  comments: Comment[];
  author: any;
  likes: any[];

  constructor(
    postId: number,
    text: string,
    imageUrl: string,
    comments: Comment[],
    author: any,
    likes: any[]
  ) {
    this.postId = postId;
    this.text = text;
    this.imageUrl = imageUrl;
    this.comments = comments;
    this.author = author;
    this.likes = likes;
  }
}

File: Project-3-Back-End-main/social-media-spring-main/src/main/java/com/yasin/models/Post.java
package com.revature.models;

import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "posts")
public class Post {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int postId;
	private String text;
	private String imageUrl;
	@OneToMany(cascade = CascadeType.ALL, mappedBy = "post", fetch = FetchType.LAZY)
	private List<Comment> comments;
	@ManyToMany(cascade = CascadeType.PERSIST)
	private List<User> likes;
	@ManyToOne
	private User author;
}
